pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -588,14 +590,27 @@ func (r *ReconcileSyncSetInstance) applySyncSetResources(ssi *hivev1.SyncSetInst
 
 	// determine if we can gather info for all resources
 	infos := []hiveresource.Info{}
+	unstructureds := []*unstructured.Unstructured{}
 	for i, resource := range resources {
+		unstructureds = append(unstructureds)
+		// Inject the hive managed annotation to help end-users see that a resource is managed by hive:
+		u := &unstructured.Unstructured{}
+		err := yaml.Unmarshal(resource.Raw, &u)
+		if err != nil {
```

Comment:
You do not need to get the reference to `u`, it is already a reference.
```suggestion
		if err := yaml.Unmarshal(resource.Raw, u); err != nil {
```